### DISCUSSION TITLE
Safelist doesn't work with sveltekit

### DISCUSSION THREAD

**Discussion by dsm-72** [tailwindlabs/tailwindcss#10345] [Help]
So I have extended my tailwindcss with 
```javascript
// tailwind.config.js
// ...
theme: {
    extend: {
      spacing: {        
        '62': '15.5rem',
        '80': '20rem',
        '84': '21rem',
        '88': '22rem',
        '92': '23rem',
        '96': '24rem',
        '100': '25rem',
        '104': '26rem',
        '108': '27rem',
        '112': '28rem',
        '116': '29rem',
        '118': '29.5rem',
        '120': '30rem',
        '124': '31rem',
        '128': '32rem',
      },
//...
```

and in a component I have

```
<script lang="ts">
export let leftMargin:number = 124;
const leftMarginStyle:string = `lg:ml-${leftMargin}`
</script>

<div class="{leftMarginStyle:string}">
</div>
```

which shows in the client `<div class="lg:ml-124"></div>` as desired... **but** the class isnt applied! However, if I manually set `<div class="lg:ml-124">` in the _component_ it works. If I go to the browser and manually set margins, my custom sizes dont work. What did I do wrong? The class is being bound correctly, but tailwind isnt catching it?

Someone on discord recommend [content configuration](https://tailwindcss.com/docs/content-configuration#safelisting-classes)

and I tried both of the following:

```javascript
// tailwind.config.js
const spacing = `62 80 84 88 92 96 100 104 108 112 116 118 120 124 128`.split(' ')
const safeSpacing = 'w h l r t b x y ml mr mt mb mx my'.split(' ')
  .map(p=>spacing.map(s=>`${p}-${s}`))
  .flat()
const mediaSpacing = 'xs sm md lg xl'.split(' ')
  .map(m=>safeSpacing.map(s=>`${m}:${s}`))
  .flat()

// ...

{
  safelist: [...safeSpacing, ...mediaSpacing],
}
```

and 

```javascript
// tailwind.config.js
// ...
safelist: [
        {
          pattern: /(w|h|l|r|t|b|x|y)-(62|80|84|88|92|96|100|104|108|112|116|118|120|124|128)/,
          variants: ['lg',],
        },
],
// ...
```

Please help me get this to work :sob:

**Comment by wongjn** [comment]
Dynamic classes like `lg:ml-${leftMargin}` indeed do not work, see https://tailwindcss.com/docs/content-configuration#dynamic-class-names.

For the safelist configurations, either work in Tailwind Play [here](https://play.tailwindcss.com/JtaOM1zipU?file=config) and [here](https://play.tailwindcss.com/GUBM1R6tBS?file=config) so it seems like it could be some misconfiguration elsewhere, or perhaps a dev-server restart might help.

As an aside, I would avoid safelisting like this – you're introducing so many classes that you may not be using. Consider avoiding string interpolation to build class names instead.
